### PR TITLE
add kubernetes section

### PR DESCRIPTION
**Pull request by loiskilberg-da**
Add a section under Deploy Daml for a guide on Kubernetes. This PR is to convert existing files into the docs.daml.com repo. Later, a language review will need to be done.

### CODE REVIEW

File: docs/2.8.0/docs/kubernetes/02-provision-cloud-resources/01-provision-cloud-resources.rst [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+* You need to provide an existing Azure resource group, you can create a new one using this command:
+
+  .. code-block:: bash
+
+     az group create --name <resource_group_name> --location <location>
+
+* Create your own backend configuration
+
+* Copy and customize variables the file ``sample.tfvars``, use the resource group you just created, you can use the file name ``terraform.tfvars`` to avoid passing argument ``--var-file=/path/to/file.tfvars`` each run.
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/kubernetes/02-provision-cloud-resources/01-provision-cloud-resources.rst:76]: I copied the sample.tfvars file and put it in the docs repo.

File: docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/01-helm-deployment/02-participant-deployment.rst [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+* Verify the status of the deployment
+
+Steps
+*****
+
+1. Inspect the Helm value files generated by Terraform
+======================================================
+
+.. note::
+   The Terraform scripts were used to parameterize the Helm values. If standalone Helm deployment is done without Terraform the value file has to be customized manually. The value file can be found under ``participant.yaml``. Note that in this guide we are deploying the participant node before deploying the domain node. We may do this because we are deploying both onto the same Kubernetes cluster, and we know the `participant node's identity <https://docs.daml.com/canton/usermanual/identity_management.html#default-initialization>`_ beforehand (e.g., that its name is ``participant1``\ ). For more details, see the `relevant section in the canton-domain Helm chart documentation <https://artifacthub.io/packages/helm/digital-asset/canton-domain#bootstrap>`_. In most cases, the operators of the participant and domain nodes differ, hence, a more involved `on-boarding process <https://docs.daml.com/canton/usermanual/identity_management.html#participant-onboarding>`_ has to be used.
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/01-helm-deployment/02-participant-deployment.rst:21]: I copied the participant.yaml file and put it in the docs repo. Need to figure out how to link to it.

File: docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/01-helm-deployment/03-domain-deployment.rst [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+* Verify the status of the deployment
+
+Steps
+*****
+
+1. Inspect the Helm value files generated by Terraform
+======================================================
+
+.. note::
+   The Terraform scripts were used to parameterize the Helm values. If standalone Helm deployment is done without Terraform, the value file has to be customized manually. The value file can be found under ``domain.yaml``.
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/01-helm-deployment/03-domain-deployment.rst:21]: I copied the domain.yaml file and put it in the docs repo. Need to figure out how to link to it.

File: docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/01-helm-deployment/04-daml-http-json-deployment.rst [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+################################
+
+Steps
+*****
+
+1. Parameterize the Helm chart
+==============================
+
+.. note::
+   The Terraform scripts were used to parameterize the Helm values. If standalone Helm deployment is done without Terraform the value file has to be customized manually. The template value file can be found in ``http-json.yaml``.
```
Comment (loiskilberg-da) [inline docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/01-helm-deployment/04-daml-http-json-deployment.rst:14]: I copied the http-json.yaml file and put it in the docs repo. Need to figure out how to link to it.

**Comment by loiskilberg-da** [comment]
More items to be resolved:

- [ ] The following link in the original doc brings up a 404 error, so I removed this sentence containing the link. Does it need to be fixed and added back to the doc?
"* Access to the :doc:`Daml Enterprise Artifactory <digitalasset-docker.jfrog.io/canton-enterprise>` repository."

- [ ] There were links in the original doc to the following repos. Is this ok?
https://legendary-fishstick-vrp5wmn.pages.github.io/src/terraform/sample.tfvars
https://github.com/DACH-NY/0-to-k8-canton-doc-temp-space/tree/main/src/helmfile 

- [ ] There is a link to a topic in the original repo that is not included in this initial release:
https://legendary-fishstick-vrp5wmn.pages.github.io/06-advanced-topics/01-azure-inventory-provisioned-resources.html For now, I removed that sentence. Let me know if that needs to be added.

File: docs/2.8.0/docs/kubernetes/01-introduction/overview.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+Deploy on Kubernetes
+####################
+
+Audience
+********
+
+This content is targeted primarily for individuals with the following skills & experience:
+
+* Basic hands-on experience with deploying and managing applications running on `Kubernetes <https://kubernetes.io/>`_.
+* Deployed applications packaged as `Helm charts <https://helm.sh/>`_.
```
Comment (carrielaben-da) [inline docs/2.8.0/docs/kubernetes/01-introduction/overview.rst:13]:
```suggestion
* Deploying applications packaged as `Helm charts <https://helm.sh/>`_.
```

File: docs/2.8.0/docs/kubernetes/02-provision-cloud-resources/01-provision-cloud-resources.rst [hunk 6]
```diff
@@ -0,0 +1,268 @@
+.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Provisioning Cloud Resources with Terraform
+###########################################
+
+.. note::
+   All the resources created will be private, nothing is being exposed over the public internet.
```
Comment (carrielaben-da) [inline docs/2.8.0/docs/kubernetes/02-provision-cloud-resources/01-provision-cloud-resources.rst:8]:
```suggestion
   All the resources created are private, nothing is being exposed over the public internet.
```

File: docs/2.8.0/docs/kubernetes/images/k8s-deployment.png
Comment (carrielaben-da) [inline docs/2.8.0/docs/kubernetes/images/k8s-deployment.png:1]: The references to "domain topology manager" in this diagram should be either shortened to "topology manager" or expanded to "sync domain topology manager"

**Review by carrielaben-da — APPROVED** [review]
Looking good so far, I saved the in-depth stuff for later except where I couldn't help myself.

File: docs/2.8.0/docs/kubernetes/images/k8s-deployment.png
Comment (loiskilberg-da) [reply docs/2.8.0/docs/kubernetes/images/k8s-deployment.png:1]: I can send it to Jennifer and Preksha to see if they can edit it.

**Review by miklos-da — APPROVED** [review]
Looks good — we should be able to fix all references to `.yaml` and other files once we have the open-source repo set up for the accompanying content for the guide.

File: docs/2.8.0/docs/kubernetes/images/k8s-deployment.png
Comment (miklos-da) [reply docs/2.8.0/docs/kubernetes/images/k8s-deployment.png:1]: I fixed this in our internal repo and added a story to our backlog to change wherever else needed. Can you point me to the guidelines you have on nomenclature around Daml Enterprise/Canton components that we should be following, please?

Comment (loiskilberg-da) [reply docs/2.8.0/docs/kubernetes/images/k8s-deployment.png:1]: I will add the updated diagram in the next commit. @miklos-da I will send you a link on Slack to our latest glossary for terminology.

File: docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/02-helmfile-deployment/00-intro.rst [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+  .. code-block:: bash
+
+     cd src/helmfile/
+     helmfile apply --skip-diff-on-install
+
+Manual Deployment
+*****************
+
+You can go to `src/helmfile/ <https://github.com/DACH-NY/0-to-k8-canton-doc-temp-space/tree/main/src/helmfile>`_ and inspect `helmfile.yaml <https://github.com/DACH-NY/0-to-k8-canton-doc-temp-space/blob/main/src/helmfile/helmfile.yaml>`_ and its `values <https://github.com/DACH-NY/0-to-k8-canton-doc-temp-space/tree/main/src/helmfile/values>`_.
```
[the remark this answers is not in the document]
Comment (loiskilberg-da) [reply docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/02-helmfile-deployment/00-intro.rst:42]: will remove in next commit

File: docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/03-argocd-deployment/02-all-in-one-argocd-deployment.rst [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+Steps
+*****
+
+1. Set Up Repo for Argo CD
+==========================
+
+.. code-block:: bash
+
+   argocd repo add https://github.com/DACH-NY/0-to-k8-canton-doc-temp-space.git "${ARGO_ARGS[@]}" --username (USER) --password (PASS)
```
[the remark this answers is not in the document]
Comment (loiskilberg-da) [reply docs/2.8.0/docs/kubernetes/04-deploy-daml-enterprise/03-argocd-deployment/02-all-in-one-argocd-deployment.rst:20]: will remove in next commit

**Pull request merged**